spec/lang/locks.md

```diff
@@ -0,0 +1,183 @@
+# Machine Locks
+
+This file describes how system locks work in MiniRust.
+This does not match the actual lock implementations in Rust, it serves more as a specification for idealized locks.
+
+## The Lock State
+
+```rust
+pub enum LockState {
+    Unlocked,
+    LockedBy(ThreadId),
+}
+
+type LockId = Int;
+```
+
+We implement locks in the thread manager, since they are mostly used to synchronize between threads.
+
+```rust
+impl<M: Memory> ThreadManager<M> {
+    pub fn lock_create(&mut self) -> LockId {
+        let id = self.locks.len();
+
+        self.locks.push(LockState::Unlocked);
+
+        id
+    }
+
+    pub fn lock_acquire(&mut self, lock_id: LockId) -> Result {
+        let active = self.active_thread.unwrap();
+
+        let Some(lock) = self.locks.get(lock_id) else {
+            throw_ub!("acquiring non existing lock");
+        };
+
+        // If the lock is not taken the lock is acquired by the active thread.
+        // Otherwise the thread gets blocked.
```

Comment:
```suggestion
        // If the lock is not taken, the lock gets acquired by the current (active) thread.
        // Otherwise, the active thread gets blocked.
```